### PR TITLE
Wave 12

### PR DESCRIPTION
**Pull request by laerreal**
Patches from `devel` with device boilerplate customization features (mostly).

# v2
- no empty after each `memset` in system bus device reset handler
- coding style

### CODE REVIEW

File: qemu/sysbusdevice.py [hunk 1]
```diff
@@ -214,12 +215,26 @@ def fill_source(self):
...
 
-            if reg.reset is not None:
-                reg_resets.append("s->%s@b=@s%s;" % (
-                    qtn.for_id_name,
-                    reg.reset.gen_c_code()
-                ))
-                s_is_used = True
+            if reg.size > 8:
+                if reg.reset is not None:
+                    reg_resets.append("memset(s->%s,@s%s,@s0x%x);\n" % (
```
Comment (BDanAnd) [inline qemu/sysbusdevice.py]: Нужна ли пустая строка после каждого memset?

File: qemu/register.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    OpNot,
+    OpOr,
+    OpSDeref,
+    OpSub,
+    StrConcat,
+    SwitchCase,
+)
+
+from math import (
+    log
```
Comment (BDanAnd) [inline qemu/register.py]: запятая?

File: qemu/register.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+        """
+        super(Register, self).__init__(size, name)
+        self.access = access
+
+        if reset is _undefined:
+            if virtual:
+                reset = None
+            else:
+                # Non-virtual register is normally always reset.
+                # But somtimes a device has a register that only resetted under
```
Comment (BDanAnd) [inline qemu/register.py]: sometimes

File: qemu/model_state.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+            if vms_macro_name is None:
+                raise Exception(
+                    "VMState generation for type %s is not implemented" % (
+                        f_c_type_name,
+                    )
+
+                )
+
+
```
Comment (BDanAnd) [inline qemu/model_state.py]: 2 пустые строки?

File: qemu/model_state.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                if f.array_size is not None:
+                    vms_macro_name += "_ARRAY"
+                    fdict["_n"] = str(f.array_size)
+
+            if vms_macro_name is None:
+                raise Exception(
+                    "VMState generation for type %s is not implemented" % (
+                        f_c_type_name,
+                    )
+
```
Comment (BDanAnd) [inline qemu/model_state.py]: пустая строка?

**Review by BDanAnd — COMMENTED** [review]
v1 viewed

**Comment by laerreal** [comment]
v2 is ready

**Pull request merged**